sw/device/lib/base/mmio.h

```diff
@@ -13,7 +19,12 @@
  * A reg32_t is an opaque handle to an MMIO register; it should only be modified
  * using the functions provided in this header.
  */
-typedef struct reg32 { volatile uint32_t *inner_ptr; } reg32_t;
+typedef struct reg32 {
+  // NOTE: we define the inner_ptr as an address, rather than as a pointer,
+  // since test code may want to stash a non-volatile pointer to a test fixture
+  // here, instead of a pointer to a real MMIO device.
+  uintptr_t inner_ptr;
```

Comment:
I guess this change is required only for the `reinterpret_cast<>` in `mock_mmio.c`, right?

Reply:
Yup. We just need to be careful that the same code doesn't try to reinterpret this as different types.